### DISCUSSION TITLE
v0.5.0

### DISCUSSION THREAD

**Discussion by anacrolix** [anacrolix/cove#36] [Announcements]
See the [changelog](https://coveapp.info/CHANGELOG#v0.5.0), and also the [Docker image](https://hub.docker.com/r/anacrolix/cove).

<hr /><em>This discussion was created from the release <a href='https://github.com/anacrolix/cove/releases/tag/v0.5.0'>v0.5.0</a>.</em>

**Comment by Photopuppet** [comment]
Thank you for the new release! Unfortunately on Windows, no luck running it... I get this error message (with a fresh install or upgraded the old installation) on two PCs:

E:\cove>run-console

E:\cove>cove.exe    2>&1  | tee-x64.exe cove.log
[2024-11-22 13:38:30 +0000 CRT github.com/anacrolix/gostdapp app.go:47]
  msg="error in main: initializing local DHT indexer: creating search conn factory: opening caterwaul db: pebble: merger name from file \"dht-indexer-bitmap-merger\" != merger name from options \"caterwaul-merger\""

E:\cove>

**If I delete the Cove appdata folder so like a 'fresh' install, then I get this message:**

E:\cove>cove.exe    2>&1  | tee-x64.exe cove.log
[2024-11-22 13:44:48 +0000 WRN github.com/anacrolix/dht-indexer/sqlite/datadb open.go:110]
  msg="upgrading database" "from version"=0 "to version"=1
panic: open schemas\torrents\1\base-tables.sql: file does not exist [recovered]
        panic: open schemas\torrents\1\base-tables.sql: file does not exist [recovered]
        panic: open schemas\torrents\1\base-tables.sql: file does not exist

goroutine 1 [running, locked to thread]:
cove.MainErr.func3()
        cove/main.go:242 +0x353
panic({0x7ff68be1be40?, 0xc00048f020?})
        runtime/panic.go:785 +0x132
cove.MainErr.func4()
        cove/main.go:260 +0x205
panic({0x7ff68be1be40?, 0xc00048f020?})
        runtime/panic.go:785 +0x132
github.com/anacrolix/dht-indexer/sqlite.UnwrapBytes(...)
        github.com/anacrolix/dht-indexer/sqlite/sqlite.go:35
github.com/anacrolix/dht-indexer/sqlite.EmbeddedScriptQueries({0xc00060c300?, 0x7ff68c5d53f0?})
        github.com/anacrolix/dht-indexer/sqlite/sqlite.go:41 +0x6e
github.com/anacrolix/dht-indexer/sqlite.ExecuteEmbeddedSqliteScriptsFancyCtx({0x7ff68c5d53f0, 0xc00015c5a0}, 0xc000073c78, 0x0, {0xc000603ab0, 0x1, 0x4?})
        github.com/anacrolix/dht-indexer/sqlite/sqlite.go:93 +0x1b7
github.com/anacrolix/dht-indexer/sqlite/datadb.upgradeUserVersionWithEmbeddedScripts.func1.1.1()
        github.com/anacrolix/dht-indexer/sqlite/datadb/open.go:61 +0x2d
github.com/anacrolix/dht-indexer/sqlite/datadb.runIfVersion({0x7ff68c5d53f0, 0xc00015c5a0}, {{0x7ff68c5d53f0?, 0xc00015c5a0?}, 0xc000073c78?, 0xc00056a390?}, 0x0, 0xc00093e5a8)
        github.com/anacrolix/dht-indexer/sqlite/datadb/open.go:111 +0x2fc
github.com/anacrolix/dht-indexer/sqlite/datadb.upgradeUserVersionWithEmbeddedScripts.func1.1()
        github.com/anacrolix/dht-indexer/sqlite/datadb/open.go:60 +0xaa
github.com/go-llsqlite/adapter/sqlitex.WithTransactionRollbackOnError(0xc000073c78, {0x7ff68c0523a4?, 0xc000a821a0?}, 0xc00093e6e8)
        github.com/go-llsqlite/adapter@v0.1.1-0.20240830122034-c3f82647e9e7/sqlitex/sqlitex.go:14 +0xbd
github.com/anacrolix/dht-indexer/sqlite/datadb.upgradeUserVersionWithEmbeddedScripts.func1({{0x7ff68c5d53f0?, 0xc00015c5a0?}, 0xc000073c78?, 0xc00056a390?})
        github.com/anacrolix/dht-indexer/sqlite/datadb/open.go:56 +0xd4
github.com/anacrolix/dht-indexer/fancydb.(*DbConn).DoFancy(0x12?, {0x7ff68c5d53f0, 0xc00015c5a0}, 0xc00093e800)
        github.com/anacrolix/dht-indexer/fancydb/fancydb.go:109 +0x70
github.com/anacrolix/dht-indexer/sqlite/datadb.upgradeUserVersionWithEmbeddedScripts({0x7ff68c5d53f0, 0xc00015c5a0}, 0x0?, 0x0, {0xc000603ab0?, 0xc00009b380?, 0x1?})
        github.com/anacrolix/dht-indexer/sqlite/datadb/open.go:55 +0xb1
github.com/anacrolix/dht-indexer/sqlite/datadb.RunDatabaseUpgrades(...)
        github.com/anacrolix/dht-indexer/sqlite/datadb/open.go:75
github.com/anacrolix/dht-indexer/sqlite/datadb.initDataDb({0x7ff68c5d53f0, 0xc00015c5a0}, 0xc0005c8d20)
        github.com/anacrolix/dht-indexer/sqlite/datadb/open.go:84 +0x18a
github.com/anacrolix/dht-indexer/sqlite/datadb.OpenDataDb({0x7ff68c5d53f0, 0xc00015c5a0}, {0xc0001d70c0, 0x33}, 0x0)
        github.com/anacrolix/dht-indexer/sqlite/datadb/open.go:35 +0x325
github.com/anacrolix/dht-indexer.(*caterwaulSearchConnFactory).Open(0xc000940900, {0x7ff68c5d53f0, 0xc00015c5a0}, 0x0)
        github.com/anacrolix/dht-indexer/search-conn.go:55 +0x52
cove.initSearchDbMaintainer({0x7ff68c5d53f0, 0xc00015c5a0}, 0xc0003f5700, {0x7ff68c5cc378, 0xc000940900}, {{0x1, {0x0, 0x0, 0x0}, {0x0, ...}, ...}})
        cove/main.go:975 +0x59
cove.initLocalDhtIndexer({_, _}, {{_, _}, {_, _}, {_, _}, _}, {0x0, ...}, ...)
        cove/main.go:859 +0x8a5
cove.MainErr({0x7ff68c5d51e8, 0xc0003f5540})
        cove/main.go:487 +0x2865
github.com/anacrolix/gostdapp.RunContext(0x7ff68c3e6e68)
        github.com/anacrolix/gostdapp@v0.1.1-0.20240928112503-bc494d64e1e4/app.go:26 +0xf1
main.main()
        cove/cove/main_else.go:14 +0x1a

E:\cove>

Many thanks for taking a look!

**Reply by anacrolix** [reply]
Thanks very much for reporting this. It should be fixed in v0.5.1. https://github.com/anacrolix/cove/releases/tag/v0.5.1
